src/compile/scale/parse.ts

```diff
@@ -53,8 +55,11 @@ export function parseScale(model: UnitModel, channel: Channel) {
     range: parseRange(scale)
   };
 
+  // See comment in selection.ts for why such a hack is necessary.
```

Comment:
This comment won't be clear for reader what is the hack here. 

Maybe add a comment right above the signal line to say that the signal there is not a proper signal name but r rather a temporary name that will be replaced later (and see comment for `isRawSelectionDomain` in selection.ts for more details)?